src/python/pants/backend/python/goals/export.py

```diff
@@ -3,50 +3,83 @@
 
 from __future__ import annotations
 
+import logging
 import os
+from collections import defaultdict
 from dataclasses import dataclass
+from typing import DefaultDict
 
 from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.target_types import PythonResolveField
 from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
 from pants.backend.python.util_rules.pex import VenvPex, VenvPexProcess
 from pants.backend.python.util_rules.pex_environment import PexEnvironment
 from pants.backend.python.util_rules.pex_from_targets import RequirementsPexRequest
-from pants.core.goals.export import ExportableData, ExportableDataRequest, ExportError, Symlink
-from pants.engine.internals.selectors import Get
+from pants.core.goals.export import ExportError, ExportRequest, ExportResult, ExportResults, Symlink
+from pants.core.util_rules.distdir import DistDir
+from pants.engine.engine_aware import EngineAwareParameter
+from pants.engine.internals.selectors import Get, MultiGet
 from pants.engine.process import ProcessResult
 from pants.engine.rules import collect_rules, rule
+from pants.engine.target import Target
 from pants.engine.unions import UnionRule
+from pants.util.docutil import bin_name
+from pants.util.strutil import path_safe
+
+logger = logging.getLogger(__name__)
 
 
 @dataclass(frozen=True)
-class ExportedVenvRequest(ExportableDataRequest):
+class ExportVenvsRequest(ExportRequest):
     pass
 
 
+@dataclass(frozen=True)
+class _ExportVenvRequest(EngineAwareParameter):
+    resolve: str | None
+    root_python_targets: tuple[Target, ...]
```

Comment:
The word "root" here seems to be more confusing than useful? AFAICT there is nothing here that requires these to be "root targets" in the usual sense?

Reply:
They're the input targets that you specified to run on, as compared to their dependencies. This change is in line with https://github.com/pantsbuild/pants/pull/14323